libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncDownTarget.java

```diff
@@ -165,27 +168,48 @@ public int cleanGhosts(SyncManager syncManager, String soupName) throws JSONExce
         return localIdSize;
     }
 
+    /**
+     * Return predicate to target records with this sync id if there is an index on __sync_id__
+     * @param syncManager
+     * @param soupName
+     * @param syncId
+     * @return
+     */
+    protected String buildSyncIdPredicateIfIndexed(SyncManager syncManager, String soupName, long syncId) {
+        String additionalPredicate = "";
+        IndexSpec[] indexSpecs = syncManager.getSmartStore().getSoupIndexSpecs(soupName);
+        for(IndexSpec indexSpec : indexSpecs) {
+            if (indexSpec.path.equals(SYNC_ID)) {
+                additionalPredicate = String.format("AND {%s:%s} = %d", soupName, SYNC_ID, syncId);
```

Comment:
<table data-meta="generated_by_SmartSync">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/51a9f8116cbddb494a8bc52842ad3f2f29c2fbd9/libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncDownTarget.java#L183"></span>Implicitly using the default locale is a common source of bugs: Use <code>String.format(Locale, ...)</code> instead</td>
    </tr>
  </tbody>
</table>